src/modules/returns/lib/helpers.js

```diff
@@ -96,6 +94,30 @@ const getLicenceReturns = async (licenceNumbers, page = 1, isInternal = false) =
   return { data, pagination };
 };
 
+/**
+ * Checks whether user uses XML Upload for a list of licence numbers
+ * @param {Array} licenceNumbers to check if isUpload flag is true
+ * @param {Boolean} isInternal to specify user type
+ * @return {Boolean} if user has XML Upload functionality
+ */
+const isXmlUpload = async (licenceNumbers) => {
+  const filter = {
+    'metadata->>isUpload': 'true',
+    'licence_ref': { '$in': licenceNumbers }
+  };
+
+  const requestPagination = { 'page': 1, 'perPage': 1 };
+  const columns = ['return_id'];
+
+  const { error, pagination } = await returns.findMany(filter, {}, requestPagination, columns);
+  if (error) {
```

Comment:
We could shorten this a little if we imported `throwIfError` from hapi-pg-rest-api:
```
throwIfError(error);
return pagination.totalRows > 0
```